src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignedCms.cs

```diff
@@ -470,5 +470,63 @@ public void CheckHash()
         {
             return ref _signedData;
         }
+
+        public void AddCertificate(X509Certificate2 certificate)
+        {
+            int existingLength = _signedData.CertificateSet?.Length ?? 0;
+
+            byte[] rawData = certificate.RawData;
```

Comment:
Seems like this should be inside of the below if? (And if it's likely CertificateSet will be empty, ideally it'd further only happen if there's at least one cert).

Reply:
@stephentoub, actually I meant to reuse this variable when creating `CertificateChoiceAsn`. Will fix it